torch/csrc/autograd/saved_variable.cpp

```diff
@@ -43,6 +43,18 @@ SavedVariable::SavedVariable(const Variable& variable, bool is_output, bool is_i
     saved_version_ = version_counter.current_version();
     is_leaf_ = variable.is_leaf();
     is_output_ = is_output;
+    is_inplace_on_view_ = is_inplace_on_view;
+
+    if(is_inplace_on_view) {
+      TORCH_INTERNAL_ASSERT(!is_leaf_ && is_output);
+      weak_grad_fn_ = variable.grad_fn();
+    }
```

Comment:
The previous comment was `// From now on, we can assume the variable is not a leaf and is an output.`. Why was it fine to move this code up from where it used to be?

Reply:
The short answer is:
A saved variable can be an `is_inplace_on_view` only if it `is_output` and not `is_leaf`.
If `is_inplace_on_view` is true, we will never run into the code about `saved_original`, so it's fine.
If `is_inplace_on_view` is ƒalse, that snippet of code was and still is a no op.

Reply:
Hmm, taking a step back: what does "is_inplace_on_view" mean in the first place? I can understand what it means for a SavedVariable to be an output, but if "is_inplace_on_view" is True, what does that mean for the SavedVariable?

Reply:
Does it mean that the SavedVariable holds a Tensor that is the output of the in-place operation on the view?

Reply:
I read through some more autograd code and convinced myself that this is the case :)

Reply:
Yes that's exactly correct. An example is 
```
a = torch.randn(6, requires_grad=True)
b = a.clone().view(3, 2)
b.relu_()
```

The relevant call to the `SavedVariable` constructor is https://github.com/pytorch/pytorch/blob/4c3eea26bd80c7126f41ff51d0dcc32f5aae5567/tools/autograd/gen_variable_type.py#L655-L658